File: app/src/main/java/in/obvious/ws/writingkotlin/lambdas/java/Bank.java
package in.obvious.ws.writingkotlin.lambdas.java;

import java.util.ArrayList;

public class Bank {

    private final ArrayList<Account> accounts;

    private final OnAccountAddedListener accountAddedListener;

    public Bank(ArrayList<Account> accounts, OnAccountAddedListener accountAddedListener) {
        this.accounts = accounts;
        this.accountAddedListener = accountAddedListener;
    }

    public void addAccount(Account account) {
        accounts.add(account);

    }

    interface OnAccountAddedListener {
        void onAccountAdded(Account account);
    }
}

File: app/src/main/kotlin/in/obvious/ws/writingkotlin/lambdas/kotlin/Bank.kt
package `in`.obvious.ws.writingkotlin.lambdas.kotlin

import `in`.obvious.ws.writingkotlin.lambdas.java.Account

class Bank(
    private val accounts: MutableList<Account>,
    private val accountAddedListener: (Account) -> Unit
) {

    fun addAccount(account: Account) {
        accounts.add(account)
        accountAddedListener(account)
    }
}
